### ISSUE TITLE
bug: Parameterised route not create new instance of view

### ISSUE THREAD

**Issue by hiddenboox** [ionic-team/ionic-framework#28509]
### Prerequisites

- [X] I have read the [Contributing Guidelines](https://github.com/ionic-team/ionic-framework/blob/main/.github/CONTRIBUTING.md#creating-an-issue).
- [X] I agree to follow the [Code of Conduct](https://ionicframework.com/code-of-conduct).
- [X] I have searched for [existing issues](https://github.com/ionic-team/ionic-framework/issues) that already report this problem, without success.

### Ionic Framework Version

v7.x

### Current Behavior

```   
I have simple routing:
<IonApp>
  <IonReactRouter>
    <IonRouterOutlet>
      <Route path="/" exact component={Home} />
      <Route path="/post/:postId" exact component={Post} />
    </IonRouterOutlet>
  </IonReactRouter>
</IonApp>
```

I suppose that one Post should be able to reference to another post in the same route eg `/post/:id` 
Post 1 `/post/1` navigates to Post 2`/post/2` then I want to swipe back so during swiping back I should see cached view from Post 1 but current I see duplicated view. 

### Expected Behavior

I expect to have two cached view one for `/post/1` and second for `/post/2` so during swiping back from `/post/2` I will see cached view of `/post/1`.

### Steps to Reproduce

1. pnpm install
2. pnpm dev
3. open localhost:5173
4. click on Post 1
5. click on Post 2
6. swipe back 
7. you also can set animation slowdown on google chrome dev tools to see that Post 1 are cloned from Post 2 cuz Post 1 was override by Post 2

### Code Reproduction URL

https://github.com/hiddenboox/ionic-react-repro

### Ionic Info

Ionic:
Ionic CLI : 7.1.1

Utility:

   cordova-res : not installed globally
   native-run  : not installed globally

System:

   NodeJS : v16.17.1
   npm    : 8.15.0
   OS     : macOS Unknown

### Additional Information

https://github.com/ionic-team/ionic-framework/assets/2965120/9b5058ac-efbb-447e-b45f-104a94306478

I have provided PR for this #28507

**Comment by sean-perkins** [comment]
Hello @hiddenboox thanks for reporting this issue!

I'm going to close this as a duplicate of #26524. 

The behavior of returning the correct cached view item is fairly complex against how Ionic's react routing implementation is architected. 

I have done some of the initial work here: https://github.com/ionic-team/ionic-framework/commit/1ab433be079ae609de5e978f744fe738126b6176#diff-0044ec6901453c85bf60a16b68b2868602535c9248931de9f22aac92783d04f9R43-R51

But there is still quite a bit of refactoring to separate registering the page (which can happen on the page mount, but also as part of a transition from the stack manager) and returning the correct cached view item. 

The full discovery changes are here if you are interested: https://github.com/ionic-team/ionic-framework/compare/main...sp/react-router-param-issue

**Comment by hiddenboox** [comment]
Hey @sean-perkins, thanks for your work, btw. routing/navigation system in ionic is rly nice. If you want please check #28507 
this patch fixes problem from #26524. 

Thanks
